src/main/java/com/cloudbees/jenkins/support/filter/ContentFilters.java

```diff
@@ -44,7 +43,7 @@
 public class ContentFilters extends GlobalConfiguration {
 
     public static ContentFilters get() {
-        return ExtensionList.lookupSingleton(ContentFilters.class);
+        return GlobalConfiguration.all().get(ContentFilters.class);
```

Comment:
Do you think it's actually this and not DeadlockTrackChecker requiring this before it should? Have you tested that scenario?

Reply:
I think that the initialization of the `final ContentFilter` field in `DeadlockTrackChecker` was causing the problem.

But my findings were that `ExtensionList.lookupSingleton` is always returning an instance of the `ContentFilters` and may create the `ExtensionList` if it does not exist yet. In that case, and in the scenario of dynamic loading of the plugin, it seems to potentially cause the same extension to be loaded twice. By using `GlobalConfiguration.all()` we go through a different cause path that seems less prone to this error. I'd love that someone to confirm this.

> Have you tested that scenario?

I have only tested:
* the fix of `ContentFilters#get` alone: OK
* the fix of `ContentFilters#get` and `DeadlockTrackChecker` together: OK

I can test the fix for `DeadlockTrackChecker`  in isolation quickly.

Reply:
I don't know it exactly either but if you have the theory and it's not hurting, sounds good.

Reply:
I'm a bit concern that it may not be the fix and the tests were not failing because it's hard to reproçduce. Maybe you can try to reproduce the problem deterministically in a test even in Jenkins core, instead of the plugin itself, you could take ideas from `PluginManagerTest` (`dynamicLoad`) or `InstallPluginCommandTest`.